book/src/validator.md

```diff
@@ -0,0 +1,48 @@
+# Anatomy of a Validator
+
+## History
+
+When we first started Solana, the goal was to de-risk our TPS claims. We
+knew that between optimistic concurrency control and sufficiently long
+leader slots, that PoS consensus was not the biggest risk to TPS. It
+was GPU-based signature verification, software pipelining and concurrent
+banking. Thus, the TPU was born. After topping 100k TPS, we split the team into
+one group working toward 710k TPS and another to flesh out the validator
+pipeline. Hence, the TVU was born. The current architecture is a consequence of
+increment development with that ordering and project priorities. It is not a
+reflection of what we ever believed was the most technically elegant
+cross-section of those technologies. In the context of leader rotation, the
+strong distinction between leading and validating is blurred.
+
+## Difference between validating and leading
+
+The fundamental difference between the pipelines is when the PoH is present. In
+a leader, we process transactions, booting bad ones, and then tag the result
+with a PoH hash. In the validator, we verify that hash, peel it off, and
+process the transactions in exactly the same way. The only difference is that
+if a validator sees a bad transaction, it can't simply boot it like the leader
+does, because that would cause the PoH hash to change.  Instead, it rejects the
+whole block. The other difference between the pipelines is what happens *after*
+banking. The leader broadcasts entries to downstream validators whereas the
+validator will have already done that in RetransmitStage, which is a
+confirmation time optimization.  The validation pipeline, on the other hand,
+has one last step. Any time it finishes processing a block, it needs to weigh
+any forks it's observing, possibly cast a vote, and if so, reset its PoH hash
```

Comment:
Should we show voting in the diagram? Or, is that signified by the array from Retransmit Stage to Gossip Service?

Reply:
Done, thanks!